auth/store.go

```diff
@@ -887,7 +864,7 @@ func (as *authStore) isAuthEnabled() bool {
 	return as.enabled
 }
 
-func NewAuthStore(be backend.Backend, indexWaiter func(uint64) <-chan struct{}) *authStore {
+func NewAuthStore(be backend.Backend, indexWaiter func(uint64) <-chan struct{}, tokenType, tokenOpts string) *authStore {
```

Comment:
I see. I'll decouple the construction of token provider from the construction of auth store.